ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,98 @@
+
+A Workspace is an abstraction of a namespace. Each workspace has a single namespace
+which backs it.
+
+Workspaces are required by [kubesaw] for providing a transparent multi-cluster communication for us. A user's workspace may be allocated to one cluster or another - but our proxy layer keeps the user blissfully unaware of this. Workspaces can be placed on clusters either automatically (using [kubesaw] mechanisms) or manually by a platform administrator.
+
+Workspace roles give a way to describe who has what rights in the workspace.
+
+Workspaces happen to be our mechanism for managing quota - a fact inherited from the [kubesaw] stack.
+
+### Some changes
+
+Some relevant changes that have occurred or are ongoing.
+
+* We want to make Konflux a successful open source projects that fits into the
+cloud native apps ecosystem. We want everything to work on vanilla Kubernetes - no OpenShift required.
+
+* We had our [first upstream Konflux release](https://github.com/konflux-ci/konflux-ci) which doesn't use [kubesaw] or relying
+on the Workspace abstraction, and is deployable on vanilla Kubernetes.
+
+* We made a decision (now long ago) to drop [KCP] from our architecture, which initially introduced the workspace concept. Both transparent multi-cluster as a feature and workspaces as a concept came from KCP.
+
+* We made the decision (at Red Hat) to stop trying to offer a managed service interface for Konflux. The user signup, waitlist, and workspace provisioning interfaces made a lot of sense there.
+
+## Decision
+
+1. We will remove the Workspace abstraction from Konflux and use standard Kubernetes namespaces
+directly for providing a place for users to run their pipelines and store their configurations
+and secrets.
+
+2. We will stop using the term Workspace, and start to use the term Namespace.
+
+3. The Konflux UI will expose a wizard for creating a new namespace. This
+wizard will be available to any user that has permissions to create namespaces.
+
+4. Konflux will provide an optional automation for creating a namespace for a user
+when he/she logins for the first time.
+
+5. Konflux won't be opinionated about the mechanism for initializing/maintaining namespaces with
+supporting resources such as ResourceQuota and LimitRange. Konflux will differ
+this responsibility to other tools which are specialized in this task such as 
+(but not limited to) [Kyverno](https://kyverno.io/policies/best-practices/add-ns-quota/add-ns-quota/) and [ArgoCD](https://github.com/konflux-ci/namespace-generator).
+
+6. Konflux will provide a thin [backend service](https://github.com/konflux-ci/workspace-manager) for listing the namespaces where the user has at least view access
```

Comment:
> What about delete operation, I would prefer if clients (UI, CLI etc..) don't have access to delete namespace. As you know that is another limitation, you are given access to delete all namespace or none. We need to improve the experience we have today where user have to ask admin/sres to delete a workspace.

An admin, for example, can use Kyverno to block the deletion of a namespace if the user that tries to delete it isn't the user which created it. This is a great example of task I think we should delegate to other tools, since it's not a Konflux specific issue.

Reply:
or with an abstraction the user itself could set a property in the workspace CR and let workspace controller handle the lock. This need to be self-service not require an admin everytime you want to lock or unlock a workspace. This also has a compliance implications. For example it is better to trace that operation directly to the end user to make sure there is an explicit desire for the user to delete a workspace and avoid accidental deletes.

Reply:
If the are no objections I'll close this thread.